src/terms.jl

```diff
@@ -508,7 +508,9 @@ reshape_last_to_i(i, a::AbstractMatrix) = reshape(a, size(a,1), ones(Int, i-1)..
 # table, some will be vectors, possibly some matrices
 function kron_insideout(op::Function, args...)
     args = (reshape_last_to_i(i,a) for (i,a) in enumerate(args))
-    vec(broadcast(op, args...))
+    out = broadcast(op, args...)
+    # flatten higher-dimensional output to vector
+    length(size(out)) > 1 ? vec(out) : out
```

Comment:
Are we absolutely sure the result is going to be an array? This might be safer:
```suggestion
    out isa AbstractArray && ndims(out) > 1 ? vec(out) : out
```

Reply:
The only situation I can think of (currently) is if a function term returns some non-array thing still survives the broadcasted `*` without an error and wouldn't cause a downstream error when `vcat`ed together with other vectors.

Reply:
Actually, the `isa AbstractArray` check might be better overall, since the underlying problem was trying to call `vec` on things that were not `AbstractArray` (i.e. a scalar)